src/Control/Middleware/ConfirmationMiddleware.php

```diff
@@ -81,8 +82,15 @@ public function __construct(...$rules)
      */
     protected function getConfirmationUrl(HTTPRequest $request, $confirmationStorageId)
     {
+        $url = $this->confirmationFormUrl;
+
+        if (substr($url, 0, 1) === '/') {
```

Comment:
How about using `Director::is_absolute($url)`?

Reply:
ok, I've had a look and `Director::absoluteURL` is not what we can use here.

Apart from performing a lot of unnecessary extra checks, it also produces incorrect result because of the leading slash in the URL. Here's the [self::is_root_relative_url($url)](https://github.com/silverstripe/silverstripe-framework/blob/4/src/Control/Director.php#L458) call, which makes it actually ignore the BASE_URL rather than enforcing it.

Reply:
Might need to be `Director::absoluteURL($url, Director::REQUEST)`

Reply:
yeah, unfortunately it's not going to work because of the [self::is_root_relative_url($url)](https://github.com/silverstripe/silverstripe-framework/blob/4/src/Control/Director.php#L458) call

Reply:
Agreed. 👍 

Can we flag this as an issue?

Reply:
reported here - https://github.com/silverstripe/silverstripe-framework/issues/9111

Reply:
@dnsl48 if your base url is `http://www.mysite.com/subfolder/` and `$url` is `/subfolder/page`, then you don't want the absolute url to end up as `http://www.mysite.com/subfolder/subfolder/page`. That's why the is_root_relative() is needed.

These are the same rules used by web browsers for url resolution; a leading `/` will force the url to resolve relative to the site root, but `subfolder/page` will resolve relative to either the current url, or base_url if that's set.

Reply:
> Apart from performing a lot of unnecessary extra checks, it also produces incorrect result because of the leading slash in the URL. Here's the self::is_root_relative_url($url) call, which makes it actually ignore the BASE_URL rather than enforcing it.

Where is this conformationFormUrl coming from? Maybe don't put a leading `/` on it if it's causing your issue?

Reply:
>  if your base url is http://www.mysite.com/subfolder/ and $url is /subfolder/page, then you don't want the absolute url to end up as http://www.mysite.com/subfolder/subfolder/page.

I think this is a somewhat controversial assumption, depending on a particular use case rather than a general rule. In the particular scenario of this PR I'd actually would like to end up with `http://www.mysite.com/subfolder/subfolder/page`, because this is the correct result.

Reply:
> Where is this conformationFormUrl coming from? Maybe don't put a leading / on it if it's causing your issue?

I'm not sure that would be semantically incorrect. As you mentioned above - a leading / will force the url to resolve relative to the site root, but subfolder/page will resolve relative to either the current url, or <base >. However, on the server side we don't have "documents" and as such if we treat everything as the same "document" with the <base href=SS_BASE_URL>, then we won't be able to use relative URLs there (relative to a page and not the global SS_BASE_URL).